### DISCUSSION TITLE
python version 3.11 or greater

### DISCUSSION THREAD

**Discussion by younesStrittmatter** [AutoResearch/autora#368] [Core Contributions] [answered]
Is there a reason we don't allow for python 3.11 or greater in the autora-core? In some documentations, we write python 3.8.10 or greater, but that is not the case, since we allow only for versions less then 3.11 in the aurora-core

**Comment by musslick** [comment]
Good question. I tried installing 3.8., and it automatically chose 3.8.2 which didn't work for AutoRA. When I looked it up, I think it has to be at least 8.3.10.

**Comment by younesStrittmatter** [comment, marked as answer]
@hollandjg relaxed the requirement of autora-core to >=3.8,<4

**Comment by hollandjg** [comment]
There's a risk if you're running older versions of python that there'll be bugs, but we don't track all of them well enough to say which will be relevant for you. I'd recommend installing the newest version of python 3.8 (currently 3.8.16, but that will change). 

How? You can install python versions using a bunch of different methods: if you only need one version, then just installing from python.org is fine. Most people will need several different versions, and using a manager like pyenv, asdf, and rtx, homebrew or chocolatey will allow you to have several installed in parallel.

The package itself needs at least python 3.8, but we don't depend on any python features which are newer than that. That's intentional – most data science packages support python 3.8+, and we're just keeping in step with them by requiring 3.8+. (This also means that you can use AutoRA out of the box when running on HPC clusters like Brown's Oscar, where newer versions of python aren't always available.)
